pkg/dependency/credhelper/cred_helper_binary.go

```diff
@@ -103,6 +103,38 @@ func updateConfigFile(bin *credhelperbin) error {
 	return nil
 }
 
+func (bin *credhelperbin) binaryInstalled() (bool, error) {
+	return binaryInstalled(bin)
+}
+
+func (bin *credhelperbin) configFileInstalled() (bool, error) {
+	cfgPath := fmt.Sprintf("%s%s", bin.hcfg.finchPath, "config.json")
+	binCfgName := bin.credHelperConfigName()
+
+	if fileExists, err := afero.Exists(bin.fs, cfgPath); err != nil {
+		return false, err
+	} else if !fileExists {
+		return false, nil
+	}
+
+	fileRead, err := bin.fs.Open(cfgPath)
+	if err != nil {
+		return false, err
+	}
+	bytes, err := afero.ReadAll(fileRead)
+	if err != nil {
+		return false, err
+	}
+	var cfg configfile.ConfigFile
+	err = json.Unmarshal(bytes, &cfg)
+	if err != nil {
+		return false, err
+	}
+	credsStore := cfg.CredentialsStore
+	defer fileRead.Close() //nolint:errcheck // closing the file
```

Comment:
There are two exit points before the defer close. Should this be moved up?

Reply:
Good catch, fixed in latest revision